### PR TITLE
DHFPROD-4315: CRUD Match Steps Thresholds

### PR DESCRIPTION
**Pull request by brucean52**
#### Checklist: 
```diff
- Note: do not change the below
```

-  ##### Owner:

- [x] JIRA_ID included in all the commit messages
- [x] PR title is in the format JIRA_ID:Title
- [x] Rebase the branch with upstream
- [x] Squashed all commits into a single commit
- [x] Added Tests
  

- ##### Reviewer:

- [x] Reviewed Tests

### CODE REVIEW

**Review by Sanjeevani19 — CHANGES_REQUESTED** [review]
@brucean52 Please look into the scenarios I noticed:

https://drive.google.com/file/d/1N_SK3EWSorg7WofhKbf8cltQ7Xuzri_j/view
@ryanjdew , Please confirm if we can add two thresholds with the same name and action types

https://drive.google.com/file/d/1O0efhIjq6Wi8Q9799CKdJn4jfGwisSw_/view 
Once the threshold is moved on the sliding scale, and the page is refreshed, the threshold goes back to the starting position

https://drive.google.com/file/d/1lU3eGzr_m1-ncrLxOvYfQTqD6uI0xABC/view 
We might need some error message when the name field is empty

**Comment by ryanjdew** [comment]
@Sanjeevani19 You are correct. The name and action combinations should be unique.

**Comment by brucean52** [comment]
> @Sanjeevani19 You are correct. The name and action combinations should be unique.

> @Sanjeevani19 You are correct. The name and action combinations should be unique.

**Comment by brucean52** [comment]
> @brucean52 Please look into the scenarios I noticed:
> 
> https://drive.google.com/file/d/1N_SK3EWSorg7WofhKbf8cltQ7Xuzri_j/view
> @ryanjdew , Please confirm if we can add two thresholds with the same name and action types
> 
> https://drive.google.com/file/d/1O0efhIjq6Wi8Q9799CKdJn4jfGwisSw_/view
> Once the threshold is moved on the sliding scale, and the page is refreshed, the threshold goes back to the starting position
> 
> https://drive.google.com/file/d/1lU3eGzr_m1-ncrLxOvYfQTqD6uI0xABC/view
> We might need some error message when the name field is empty

Added the duplicate threshold name validation and error message when name was empty. I didn't see the scale change issue on refresh. 
https://drive.google.com/file/d/1fFmq67Q-RzxC__jiAMJGM3mchevm2Ei8/view

File: marklogic-data-hub-central/ui/src/components/entities/matching/threshold-modal/threshold-modal.tsx [hunk 1]
```diff
@@ -0,0 +1,412 @@
...
+
+          if (actionErrorMessage === '' && nameErrorMessage === '' && Object.keys(props.editThreshold).length === 0) {
+            let newThreshold: Threshold = {
+              thresholdName,
+              action: actionType,
+              score: 0
+            };
+
+            let newStepArtifact: MatchingStep = curationOptions.activeStep.stepArtifact;
+            let duplicateNames = newStepArtifact.thresholds.filter( threshold => threshold.thresholdName === thresholdName);
```
Comment (ryanjdew) [inline marklogic-data-hub-central/ui/src/components/entities/matching/threshold-modal/threshold-modal.tsx:143]: We need to check for duplicate names when the action is custom as well.

**Review by ryanjdew — CHANGES_REQUESTED** [review]
Thresholds with a custom action are missing the duplicate name check, but everything else is working perfectly in the testing I've done.

File: marklogic-data-hub-central/ui/src/components/entities/matching/threshold-modal/threshold-modal.tsx (hunk 1 above)
Comment (brucean52) [reply marklogic-data-hub-central/ui/src/components/entities/matching/threshold-modal/threshold-modal.tsx:143]: Fixed this and made UX change to delete threshold confirmation

**Review by Sanjeevani19 — CHANGES_REQUESTED** [review]
https://drive.google.com/file/d/1-AR2pX4dCjbMvFb3bLyzF5kfhBnYYk6O/view
@brucean52 as soon as the sliding ends the tootlip shouldn't be highlighted, Please look into this, all the remaining scenarios are working as expected

**Comment by brucean52** [comment]
> https://drive.google.com/file/d/1-AR2pX4dCjbMvFb3bLyzF5kfhBnYYk6O/view
> @brucean52 as soon as the sliding ends the tootlip shouldn't be highlighted, Please look into this, all the remaining scenarios are working as expected

Fixed this.

**Review by Sanjeevani19 — APPROVED** [review]

**Review by xnikhil08 — APPROVED** [review]

**Review by ryanjdew — APPROVED** [review]

**Pull request merged**